tidy3d/web/api/autograd/autograd.py

```diff
@@ -886,6 +886,9 @@ def _run_tidy3d(
         verbose = run_kwargs.get("verbose", False)
         upload_sim_fields_keys(run_kwargs["sim_fields_keys"], task_id=job.task_id, verbose=verbose)
     path = run_kwargs.get("path", DEFAULT_DATA_PATH)
+    if task_name.endswith("_adjoint"):
+        path_parts = path.rsplit('.', 1)
+        path = path_parts[0] + "_adjoint." + path_parts[1]
```

Comment:
For clarity I'd prefer using `os.path.splitext` here, i.e.:
```python
import os

...

name, ext = os.path.splitext(path)
path = name + "_adjoint" + ext
```

Functionally identical, but a bit easier to tell what's going on I think.

Reply:
Sure this would also work, I was just reluctant to import another library

Reply:
Python stdlib is fine :)

Reply:
Thanks @txdai . We just had another discussion about this and to be very pedantic (although throwing away readability...) we reached this conclusion:

```
from os.path import basename, dirname, join
...
    if task_name.endswith("_adjoint"):
        path_parts = basename(path).split('.')
        path = join(dirname(path), path_parts[0] + "_adjoint." + ".".join(path_parts[1:]))
```

This is because we might have extensions like `.hdf5.gz` in principle. Not currently happening but to make this a bit more future-proof.

Do you want to modify your PR with this so we can merge it?

Reply:
oh and could you please make sure to run `ruff` on the PR once before committing, the tests are currently failing due to a lint error. if you use poetry, you can just `poetry run pre-commit install` on the repo, it'll run the linter automatically before the commit

Reply:
I reformatted the code and changed to the suggested implementation. Thanks!